code/modules/mob/living/simple_animal/hostile/terror_spiders/procs.dm

```diff
@@ -0,0 +1,447 @@
+			to_chat(src, "Your Orders: <span class='notice'>defend yourself & the hive, without being aggressive </span> ")
+		else if (ai_type == 2)
+			to_chat(src, "Your Orders: <span class='danger'>do not attack anyone, not even in self-defense!</span> ")
+		to_chat(src, "------------------------")
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/BroadcastOrders()
+	var/cache_thequeen = null
+	var/cache_theempress = null
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/queen/Q in mob_list)
+		if (Q.ckey || !Q.spider_awaymission)
+			cache_thequeen = Q
+			break
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/empress/E in mob_list)
+		if (E.ckey || !E.spider_awaymission)
+			cache_theempress = E
+			break
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/S in mob_list)
+		S.DoShowOrders(1,cache_thequeen,cache_theempress)
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/IsInfected(var/mob/B)
+	if (iscarbon(B))
+		var/mob/living/carbon/C = B
+		if(C.get_int_organ(/obj/item/organ/internal/body_egg))
+			return 1
+	return 0
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/Retaliate()
+	var/list/around = view(src, 7)
+	for(var/atom/movable/A in around)
+		if(A == src)
+			continue
+		if (A in enemies)
+			// they are already our enemy
+			continue
+		if (istype(A, /mob/living/simple_animal/hostile/poison/terror_spider))
+			// we can't make enemies of other spiders. Regardless of faction checks (they are unreliable, sometimes spiders get created with no faction!)
+			continue
+		if(isliving(A))
+			var/mob/living/M = A
+			var/faction_check = 0
+			for(var/F in faction)
+				if(F in M.faction)
+					faction_check = 1
+					break
+			if(faction_check && attack_same || !faction_check)
+				enemies |= M
+				visible_message("<span class='danger'>\icon[src] [src] glares at [M]! </span>")
+				// should probably exempt people who are dead...
+		else if(istype(A, /obj/mecha))
+			var/obj/mecha/M = A
+			if(M.occupant)
+				enemies |= M
+				enemies |= M.occupant
+		else if(istype(A, /obj/spacepod))
+			var/obj/spacepod/M = A
+			if(M.occupant || M.occupant2)
+				enemies |= M
+				enemies |= M.occupant
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/H in around)
+		var/retaliate_faction_check = 0
+		for(var/F in faction)
+			if(F in H.faction)
+				retaliate_faction_check = 1
+				break
+		if(retaliate_faction_check && !attack_same && !H.attack_same)
+			H.enemies |= enemies
+	if (istype(src, /mob/living/simple_animal/hostile/poison/terror_spider/queen) || istype(src, /mob/living/simple_animal/hostile/poison/terror_spider/empress)  )
+		for (var/mob/living/simple_animal/hostile/poison/terror_spider/T in mob_list)
+			T.enemies |= enemies
+	return 0
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/msg_terrorspiders(var/msgtext)
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/T in mob_list)
+		if (T.health > 0)
```

Comment:
Might want to check `T.stat` instead, that way dead and unconscious spiders will not be notified.